controllers/openstackplaybookgenerator_controller.go

```diff
@@ -114,6 +110,89 @@ func (r *OpenStackPlaybookGeneratorReconciler) Reconcile(ctx context.Context, re
 	templateParameters := make(map[string]interface{})
 	cmLabels := common.GetLabels(instance, openstackplaybookgenerator.AppLabel, map[string]string{})
 
+	// Check if fencing data is required as well (examine ControlPlane CR)
+	// FIXME: We assume there is only one ControlPlane CR for now (enforced by webhook), but this might need to change
+	controlPlaneList := &ospdirectorv1beta1.OpenStackControlPlaneList{}
+	controlPlaneListOpts := []client.ListOption{
+		client.InNamespace(instance.Namespace),
+		client.Limit(1000),
+	}
+	err = r.Client.List(context.TODO(), controlPlaneList, controlPlaneListOpts...)
+	if err != nil {
+		_ = r.setCurrentState(instance, ospdirectorv1beta1.PlaybookGeneratorError, err.Error())
+		return ctrl.Result{}, err
+	}
+
+	if len(controlPlaneList.Items) == 0 {
+		err := fmt.Errorf("no OpenStackControlPlanes found in namespace %s", instance.Namespace)
+		_ = r.setCurrentState(instance, ospdirectorv1beta1.PlaybookGeneratorError, err.Error())
+		return ctrl.Result{}, err
+	}
+
+	// FIXME: See FIXME above
+	controlPlane := controlPlaneList.Items[0]
+
+	if controlPlane.Status.ProvisioningStatus.State != ospdirectorv1beta1.ControlPlaneProvisioned {
+		msg := fmt.Sprintf("Control plane %s VMs are not yet provisioned. Requeing...", controlPlane.Name)
+		r.Log.Info(msg)
+		err = r.setCurrentState(instance, ospdirectorv1beta1.PlaybookGeneratorWaiting, msg)
+		return ctrl.Result{RequeueAfter: time.Second * 10}, err
+	}
+
+	templateParameters["EnableFencing"] = false
+
+	if controlPlane.Spec.EnableFencing {
+		// TODO: This will likely need refactoring sooner or later
+		for roleName, roleParams := range controlPlane.Spec.VirtualMachineRoles {
+			if common.StringInSlice(roleParams.RoleName, controlplane.GetFencingRoles()) && roleParams.RoleCount == 3 {
```

Comment:
IIUC you want to address custom roles where the user provided a roles file via the tripleo-tarball-config configmap in a follow up, right?

Reply:
Yes, correct.  Will have to look into unpacking binary CM data inside the operator and parsing through it to find the potentially relevant role chunks.